src/Access/Authentication.h

```diff
@@ -52,6 +57,18 @@ class Authentication
         static const TypeInfo & get(Type type_);
     };
 
+    // A signaling class used to communicate requirements for credentials.
+    template <typename CredentialsType>
+    class Require : public Exception
```

Comment:
It does look strange. We rarely define new exception classes. Are you catching this exception somewhere?

Reply:
Yes, it is used to signal the `HTTPHandler` that it should recommend a specific authentications mechanism to the client and reject the current request.

Reply:
We normally use ErrorCodes when we need to catch and analyze exceptions.